lib/rouge/formatters/html_exercism.rb

```diff
@@ -128,12 +128,21 @@ def formatted_code(tokens, line_counter)
             formatted << end_line << start_line(line_counter)
           end
         end
-        formatted.gsub(start_line(line_counter), '')
+        adjust_final_lines formatted, line_counter
+      end
+
+      def adjust_final_lines(formatted, line_counter)
```

Comment:
This is just a patch over some fundamental parsing issues and will add extra lines to solutions in some cases.
Github won't let me comment on bits of code that have not changed here, so I might need to go back to the original PR and add annotations there.